### PR TITLE
Plugin meeting fix alpha

### PR DESCRIPTION
**Pull request by chrisadubois**

### CODE REVIEW

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/selfUtils.js [hunk 1]
```diff
@@ -15,8 +15,10 @@ const selfUtils = {};
  */
 selfUtils.parse = (self, deviceId) => {
   if (self) {
-    return {
+    const parsedSelf = {
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/locus-info/selfUtils.js]: It doesn't look like this needs assigned to a variable.

File: packages/node_modules/@webex/plugin-meetings/src/media/index.js [hunk 2]
```diff
@@ -161,11 +133,42 @@ Media.attachMedia = (mediaProperties) => {
   return PeerConnectionManager.createOffer(peerConnection);
 };
 
-Media.updateMedia = () => {
+Media.updateMedia = (mediaProperties) => {
+  const {
+    mediaDirection, audioTrack, videoTrack, shareTrack, peerConnection
+  } = mediaProperties;
+
+  // update audio Trasceiver
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/media/index.js]: `Transceiver`

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 3]
```diff
@@ -515,7 +540,7 @@ export default class Meeting extends StatelessSparkPlugin {
     // Validate if these are valid meeting object property
     // TODO: add a check to make sure the value passed in the constructor
     // is not changed by any delta event
-    LoggerProxy.logger.log(`Meeting:index->updateMeeting#Meeting updated with ${JSON.stringify(object)}`);
+    // LoggerProxy.logger.log(` ${this.name} Meeting:index->updateMeeting#Meeting updated with ${JSON.stringify(object)}`);
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: Remove?

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js [hunk 4]
```diff
@@ -1076,7 +1109,15 @@ export default class Meeting extends StatelessSparkPlugin {
     return this.audio.toggle({
       mute: true,
       self: true
-    });
+    })
+      .then(() => {
+        LoggerProxy.logger.info('meeting/index->muteAudio#Audio mute succesful.');
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: `successful`

File: packages/node_modules/@webex/plugin-meetings/src/meeting/request.js [hunk 5]
```diff
@@ -232,7 +274,7 @@ export default class MeetingRequest extends StatelessSparkPlugin {
 
     return this.request({
       method: HTTP_VERBS.PUT,
-      uri: url,
+      uri,
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/src/meeting/request.js]: 👍

File: packages/node_modules/@webex/plugin-meetings/test/integration/spec/journey.js [hunk 6]
```diff
@@ -139,13 +139,21 @@ describe('plugin-meeting', () => {
   });
 
   // Alice calls bob and bob rejects it
-  describe('Successful 1:1 meeting (including Guest)', () => {
+  describe('Successful 1:1 meeting (including Guest)', function () {
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/test/integration/spec/journey.js]: Was `function` needed for the scope of `this`?

File: packages/node_modules/@webex/plugin-meetings/test/integration/spec/space-meeting.js [hunk 7]
```diff
@@ -76,11 +76,49 @@ describe('plugin-meetings-automation', () => {
         // add .then checks for alice response, should see bob and chris member status to isInMeeting = true
         .then(() => testUtils.waitForStateChange(bob.meeting, 'JOINED'))
         .then(() => testUtils.waitForStateChange(chris.meeting, 'JOINED')));
+      // .then(() => testUtils.addMedia(bob))
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/test/integration/spec/space-meeting.js]: Remove?

File: packages/node_modules/@webex/plugin-meetings/test/unit/spec/meeting/index.js [hunk 8]
```diff
@@ -467,9 +468,9 @@ describe('plugin-meetings', () => {
           meeting.meetingFSM.join();
           meeting.meetingRequest.leaveMeeting = sinon.stub().returns(Promise.resolve({body: 'test'}));
           meeting.locusInfo.onFullLocus = sinon.stub().returns(true);
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/test/unit/spec/meeting/index.js]: Do these need changed to Promises also? If not, please comment as to why the 3 need it.

File: packages/node_modules/@webex/plugin-meetings/wdio.conf.js [hunk 9]
```diff
@@ -64,20 +64,20 @@ exports.config = {
...
+        browserName: 'chrome',
+        chromeOptions: {
+          args: [
+            '--use-fake-device-for-media-stream',
+            '--use-fake-ui-for-media-stream'
+          ]
+        },
         tunnelIdentifier: process.env.SC_TUNNEL_IDENTIFIER
       }
+      // desiredCapabilities: {
```
Comment (mattnorris) [inline packages/node_modules/@webex/plugin-meetings/wdio.conf.js]: Remove?

File: packages/node_modules/@webex/plugin-meetings/sample/app.js [hunk 10]
```diff
@@ -75,16 +78,19 @@ function connect() {
...
-          document.getElementById('ringing-call').innerHTML = 'incoming call';
-        }
-        else if (!e.ringing) {
-          document.getElementById('ringing-call').innerHTML = '';
-        }
-      });
     }
 
+    m.meeting.on('meeting:ringing', (e) => {
+      if (e.ringing) {
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/sample/app.js]: yeah the kitchen sink is dirty, needs some Dawn

File: packages/node_modules/@webex/plugin-meetings/sample/app.js [hunk 11]
```diff
@@ -474,48 +664,153 @@ document.getElementById('stop-sending-video').addEventListener('click', () => {
...
+document.getElementById('resume-recording').addEventListener('click', () => {
   if (meeting) {
-    meeting.stopReceivingVideo();
+    meeting.resumeRecording();
   }
 });
 
+document.getElementById('charting').addEventListener('click', () => {
+  const ctx = document.getElementById('myChart').getContext('2d');
+  const chart = new Chart(ctx, { // eslint-disable-line
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/sample/app.js]: yep yep!

File: packages/node_modules/@webex/plugin-meetings/sample/app.js [hunk 12]
```diff
@@ -531,7 +826,21 @@ document.getElementById('getStats').addEventListener('click', () => {
         {
           id: audioReceiverId,
           correlate: 'audio', // NECESSARY KEY
-          onData: (data) => { console.log('audioReceiver->', data); },
+          history: true,
+          onData: (data) => {
+            console.log('audioReceiver->', data);
+            const k = data.data.getData();
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/sample/app.js]: laziness cleaning the kitchen sink

File: packages/node_modules/@webex/plugin-meetings/src/analyzer/analyzer.js [hunk 13]
```diff
@@ -0,0 +1,64 @@
+import {forEach, isFinite, isArray} from 'lodash';
+
+import {ANALYSIS_STATS} from '../constants';
+
+const StatsAnalyzer = {};
+
+// {valid: true/false, failed: { key: [number] }, data: { webRtcKeyToAnalyze: { valid: true/false, reports: [ { value: number, valid: true/false, difference: number } ] } } }
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/analyzer/analyzer.js]: good to know the format

File: packages/node_modules/@webex/plugin-meetings/src/analyzer/analyzer.js [hunk 14]
```diff
@@ -0,0 +1,64 @@
...
+
+    analysis.data[property] = {valid: true, reports: []};
+    analysis.failed[property] = [];
+    let previous = {value: 0};
+    let index = 0;
+
+    for (let i = series.length - 1; i > 0; i -= 1) {
+      const singular = {};
+
+      forEach(series[i].data.getData(), (webrtcData) => { // eslint-disable-line
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/analyzer/analyzer.js]: it's more or less O(n)

series[i].data.getData will have MAXIMUM 15-20 elements in the future, and currently is about 5 items, depending on the report.

File: packages/node_modules/@webex/plugin-meetings/src/config.js [hunk 15]
```diff
@@ -1,5 +1,9 @@
 export default {
+  // TODO: this needs to be defaulted
+  // to JS_SDK and clients set it to WEB or others
+  // see https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-73167
   meetings: {
+    deviceType: 'UNKNOWN',
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/config.js]: I'll change the default to WEB

File: packages/node_modules/@webex/plugin-meetings/src/constants.js [hunk 16]
```diff
@@ -326,12 +333,15 @@ export const EVENTS = {
   CONVERSATION_URL_UPDATE: 'CONVERSATION_URL_UPDATE',
   DESTROY_MEETING: 'DESTROY_MEETING',
   REMOTE_ANSWERED_DECLINED: 'REMOTE_ANSWERED_DECLINED',
+  // TODO: move all of these to LOCUSINFO.EVENTS
```
[the remark this answers is not in the document]
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/constants.js]: the next PR

File: packages/node_modules/@webex/plugin-meetings/src/meeting/index.js (hunk 4 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/meeting/index.js]: oops

File: packages/node_modules/@webex/plugin-meetings/src/media/index.js (hunk 2 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/media/index.js]: will fix

File: packages/node_modules/@webex/plugin-meetings/test/integration/spec/journey.js (hunk 6 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/test/integration/spec/journey.js]: yeah we leave it to debug and add a name

File: packages/node_modules/@webex/plugin-meetings/wdio.conf.js (hunk 9 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/wdio.conf.js]: we may use it

File: packages/node_modules/@webex/plugin-meetings/test/unit/spec/meeting/index.js (hunk 8 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/test/unit/spec/meeting/index.js]: I'll look

File: packages/node_modules/@webex/plugin-meetings/src/locus-info/selfUtils.js (hunk 1 above)
Comment (chrisadubois) [reply packages/node_modules/@webex/plugin-meetings/src/locus-info/selfUtils.js]: removed

File: packages/node_modules/@webex/plugin-meetings/sample/app.js (hunk 10 above)
[the remark this answers is not in the document]
Comment (mattnorris) [reply packages/node_modules/@webex/plugin-meetings/sample/app.js]: Yeah, Dad Jokes 😆

File: packages/node_modules/@webex/plugin-meetings/src/config.js (hunk 15 above)
[the remark this answers is not in the document]
Comment (mattnorris) [reply packages/node_modules/@webex/plugin-meetings/src/config.js]: Thank you - we'll get a proper one going and switch over.

**Comment by mattnorris** [comment]
@arun3528 please create a branch off of this PR for new functionality so this one can be merged. Bug fixes here are fine, but new functionality creates a moving target. It's better to merge this one and new stuff later.

**Pull request closed without merging**